File: answers/C/@mykeels/16-convert-to-decimal.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int toBase10(char* value, int base);

int main() {
    while (1) {
        int n;
        char value[128];
        
        printf("Enter the value of base n: ");
        scanf("%d", &n);

        printf("Enter the base %d value you wish to convert: ", n);
        scanf("%s", value);

        printf("%s base %d = %d base 10", value, n, toBase10(value, n));
    }
    return 0;
}

int toBase10(char* value, int base) {
    int sum = 0;
    int count = 0;
    for (int i = strlen(value) - 1; i >= 0; i--) {
        sum += (value[i] - '0') * (int)pow((double)base, (double)count);
        count++;
    }
    return sum;
}

File: answers/Go/@codehakase/16-convert-to-decimal.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var (
		n     int
		value string
	)
	fmt.Println("Enter the value of base n: ")
	fmt.Scanf("%d", &n)

	fmt.Printf("Enter the base %d value you wish to convert: ", n)
	fmt.Scanf("%s", &value)

	fmt.Printf("%s base %d = %d base 10", value, n, toBase10(value, n))
}

func toBase10(value string, base int) int {
	var sum, count int
	for i := len(value) - 1; i >= 0; i-- {
		pw := math.Pow(float64(base), float64(count))
		sum += int(value[i]-'0') * int(pw)
	}
	return sum
}
